sw/host/tests/manuf/otp_ctrl/BUILD

```diff
@@ -0,0 +1,48 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("@rules_rust//rust:defs.bzl", "rust_binary")
+load(
+    "//rules:opentitan_test.bzl",
+    "OPENTITANTOOL_OPENOCD_DATA_DEPS",
+    "OPENTITANTOOL_OPENOCD_TEST_CMDS",
+    "cw310_params",
+    "opentitan_functest",
+)
+
+package(default_visibility = ["//visibility:public"])
+
+rust_binary(
+    name = "otp_ctrl",
+    srcs = [
+        "src/main.rs",
+    ],
+    deps = [
+        "//sw/host/opentitanlib",
+        "@crate_index//:anyhow",
+        "@crate_index//:structopt",
+    ],
+)
+
+opentitan_functest(
+    name = "otp_ctrl_functest",
+    srcs = ["//sw/device/silicon_creator/manuf/lib:empty_functest.c"],
+    cw310 = cw310_params(
+        bitstream = "//hw/bitstream:rom_otp_dev",
```

Comment:
once you move this functest to sw/device/silicon_creator/manuf/lib/BUILD, define a new `otp_image` similar to [this](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/manuf/lib/BUILD;drc=80627cccf544e11c8507176e781f52850f5f0551;l=323), e.g.:
```
otp_image(
    name = "otp_img_otp_ctrl_functest",
    src = "//hw/ip/otp_ctrl/data:otp_json_test_unlocked0",
    visibility = ["//visibility:private"],
)
```
and a new bitstream with:
```
bitstream_splice(
      name = "bitstream_otp_ctrl_functest",
      src = "//hw/bitstream:rom",
      data = "otp_img_otp_ctrl_functest{}"
      meminfo = "//hw/bitstream:otp_mmi",
      update_usr_access = True,
      visibility = ["//visibility:private"],
  )
```

That way the OTP image you have on the FPGA will be completely un-provisioned (i.e. all zeros), except for the LC partition which will encode the TEST_UNLOCKED0 state.

Reply:
Note, if you do this, the ROM_EXEC_EN setting will also be 0, so you should have no problem connecting the JTAG, and you shouldn't need the idle_test.c, but it doesn't hurt to also put it in place.